### PR TITLE
Update and polish educators workshop intro notebook

### PR DESCRIPTION
**Pull request by rmlarose**
- Updates to current Cirq version.
- Polishing, formatting, etc.

### CODE REVIEW

File: docs/tutorials/educators/intro.ipynb [hunk 1]
```diff
@@ -307,15 +282,13 @@
...
-    "```\n",
-    "a = cirq.NamedQubit(\"a\")\n",
-    "```\n",
+    "Next, we encounter the object `cirq.H` which is a Hadamard gate with unitary\n",
     "\n",
-    "Next we encounter of the object `cirq.H`, which is a Hadamard gate.  `cirq.H` is an instance of the `cirq.HGate` class, which itself is a subclass of `Gate` (along with other classes). \n",
-    "$$H = {1 \\over \\sqrt{2}} \\left[ \\begin{array}[cc]  & 1 & 1  \\\\ 1 & -1 \\end{array}\\right]$$\n",
-    "We can use cirq to see this unitary matrix:"
+    "$$ H = {1 \\over \\sqrt{2}} \\left[ \\begin{array}[cc]  & 1 & 1  \\\\ 1 & -1 \\end{array}\\right] . $$\n",
```
Comment (balopat) [inline docs/tutorials/educators/intro.ipynb:289]: I _think_ you need newlines to render correctly (at least [on Github](https://github.com/quantumlib/Cirq/blob/4ee947775774de0fdf0ab02553f8c53b93c161a4/docs/tutorials/educators/intro.ipynb) it doesn't render well)
```suggestion
    "$$
     H = {1 \\over \\sqrt{2}} \\left[ \\begin{array}[cc]  & 1 & 1  \\\\ 1 & -1 \\end{array}\\right]
     $$\n",
```

Comment (balopat) [reply docs/tutorials/educators/intro.ipynb]: I still see a bunch of `$$` misrenderings on Github but maybe that's github's fault?

Comment (balopat) [reply docs/tutorials/educators/intro.ipynb]: It looks good on Jupyter notebook. Let's see how it behaves on devsite

**Review by balopat — APPROVED** [review]

File: docs/tutorials/educators/intro.ipynb (hunk 1 above)
Comment (rmlarose) [reply docs/tutorials/educators/intro.ipynb]: May the tex-in-jupyter-notebook gods be in our favor 🤞

**Pull request merged**